### PR TITLE
Caleb: Return a list of links on a page by default

### PR DESCRIPTION
**Pull request by calebpeffer**
For issue #421. I just extract the links from the HTML then return them as the linksOnPageParameter.

I thought about making this optional, but I couldn't see any downside to just including them. Might adding an optional parameter just add complexity?

Would love your take @rafaelsideguide @nickscamara 

Either way, the customer needs this by EOW lets get it merged :)

### CODE REVIEW

**Review by nickscamara — COMMENTED** [review]
This looks great @calebpeffer! Could you a few tests for it? Thank you :)

**Comment by calebpeffer** [comment]
> This looks great @calebpeffer! Could you a few tests for it? Thank you :)

Sure! What type of tests would be best here? Whats the expectation for a commit like this? Unit, e2e, both? Don't really know where the testing is at since I haven't contributed in a while

File: apps/api/src/scraper/WebScraper/single_url.ts [hunk 1]
```diff
@@ -109,6 +109,38 @@ function getScrapingFallbackOrder(
   return scrapersInOrder as (typeof baseScrapers)[number][];
 }
 
+function extractLinks(html: string, baseUrl: string): string[] {
```
Comment (rafaelsideguide) [inline apps/api/src/scraper/WebScraper/single_url.ts]: @calebpeffer could you move this function a extractLinks file inside api/src/scraper/WebScraper/utils?

File: apps/api/src/scraper/WebScraper/__tests__/single_url.test.ts [hunk 2]
```diff
@@ -22,3 +23,14 @@ describe('scrapSingleUrl', () => {
...
+it('should return a list of links on the mendable.ai page', async () => {
+  const url = 'https://mendable.ai';
+  const pageOptions: PageOptions = { includeHtml: true };
+
+  const result = await scrapSingleUrl(url, pageOptions);
+
+  // Check if the result contains a list of links
+  expect(result.linksOnPage).toBeDefined();
+  expect(Array.isArray(result.linksOnPage)).toBe(true);
+  expect(result.linksOnPage.length).toBeGreaterThan(0);
```
Comment (rafaelsideguide) [inline apps/api/src/scraper/WebScraper/__tests__/single_url.test.ts:35]: what about expect(result.linksOnPage).toContain('https://www.mendable.ai/blog') ?

**Review by rafaelsideguide — CHANGES_REQUESTED** [review]

File: apps/api/src/scraper/WebScraper/single_url.ts (hunk 1 above)
Comment (calebpeffer) [reply apps/api/src/scraper/WebScraper/single_url.ts]: @calebpeffer add docs too lol

**Pull request merged**